crates/bevy_reflect/src/reflect.rs

```diff
@@ -62,48 +62,51 @@ pub enum ReflectOwned {
     Value(Box<dyn Reflect>),
 }
 
-/// A reflected Rust type.
+/// A partially reflected Rust type. See [`Reflect`] for fully reflected types.
 ///
 /// Methods for working with particular kinds of Rust type are available using the [`Array`], [`List`],
 /// [`Map`], [`Tuple`], [`TupleStruct`], [`Struct`], and [`Enum`] subtraits.
 ///
 /// When using `#[derive(Reflect)]` on a struct, tuple struct or enum, the suitable subtrait for that
-/// type (`Struct`, `TupleStruct` or `Enum`) is derived automatically.
-pub trait Reflect: Any + Send + Sync {
+/// type ([`Struct`], [`TupleStruct`] or [`Enum`]) is derived automatically.
+pub trait PartialReflect: Any + Send + Sync {
     /// Returns the [type name][std::any::type_name] of the underlying type.
     fn type_name(&self) -> &str;
 
     /// Returns the [`TypeInfo`] of the underlying type.
     ///
-    /// This method is great if you have an instance of a type or a `dyn Reflect`,
+    /// This method is great if you have an instance of a type or a `dyn PartialReflect`,
     /// and want to access its [`TypeInfo`]. However, if this method is to be called
     /// frequently, consider using [`TypeRegistry::get_type_info`] as it can be more
     /// performant for such use cases.
     ///
     /// [`TypeRegistry::get_type_info`]: crate::TypeRegistry::get_type_info
     fn get_type_info(&self) -> &'static TypeInfo;
 
-    /// Returns the value as a [`Box<dyn Any>`][std::any::Any].
-    fn into_any(self: Box<Self>) -> Box<dyn Any>;
+    /// Returns the value as a [`&dyn Reflect`](Reflect),
+    /// or [`None`] if the value does not implement it.
+    fn as_full(&self) -> Option<&dyn Reflect>;
 
-    /// Returns the value as a [`&dyn Any`][std::any::Any].
-    fn as_any(&self) -> &dyn Any;
+    /// Returns the value as a [`&mut dyn Reflect`](Reflect),
+    /// or [`None`] if the value does not implement it.
+    fn as_full_mut(&mut self) -> Option<&mut dyn Reflect>;
 
-    /// Returns the value as a [`&mut dyn Any`][std::any::Any].
-    fn as_any_mut(&mut self) -> &mut dyn Any;
+    /// Returns the value as a [`Box<dyn Reflect>`](Reflect),
+    /// or `Err(self)` if the value does not implement it.
+    fn into_full(self: Box<Self>) -> Result<Box<dyn Reflect>, Box<dyn PartialReflect>>;
```

Comment:
what about `try_as_reflect` and `try_into_reflect`?

Reply:
Ooh I think I like those better! I would just make sure it's not throwing off our terminology elsewhere (i.e. references in docs to a "full" `Reflect`) since this is kinda where that comes from.